programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -198,16 +297,48 @@ static void on_event(const base_event *ev) {
         case TunnelEvent_ContextEvent: {
             const ziti_ctx_event *zev = (ziti_ctx_event *) ev;
             ZITI_LOG(INFO, "ztx[%s] status is %s", ev->identifier, zev->status);
+
+            identity_event id_event = {
+                    .Op = "identity",
+                    .Action = EVENT_ADDED,
+                    .Identifier = ev->identifier
+            };
+
+            tunnel_identity *tnl_identity = NULL;
+            if (zev->code == ZITI_OK) {
+                if (zev->identity) {
+                    id_event.Id = get_tunnel_identity(zev->identity);
+                }
+            }
+
+            size_t json_len;
+            char *json = identity_event_to_json(&id_event, MODEL_JSON_COMPACT, &json_len);
+            send_events_message(json, json_len);
+            // free(json);
```

Comment:
free json is causing error. So I commented it

Reply:
It is freed in the on_write_event method which is the callback method of the uv_write function in the send_events_message

Reply:
just remove these commented out `free()`s

Reply:
I removed the function to free from the callback method of on_write_event and uncommented free(json) here. Please review again

Reply:
you can't free until write is completed, right? you need to remove these free() calls

Reply:
done. Please review